### ISSUE TITLE
Java ou Kotlin para começar em desenvolvimento para Android

### ISSUE THREAD

**Issue by JoseCage** [training-center/forum#26]
Saudações caros devs. Eu pretendo começar a estudar focado em desenvolvimento mobile, concretamente apps **Android** (alias, segundo começará a curso de desenvolvimento Android) e diante disso me veio a dúvida de qual escolher entre Java ou **Kotlin**, uma vez que ainda não tenho conhecimento "suficiente" em nenhuma das duas linguagens. Praticamente eu estarei tendo o primeiro contacto com o **Java** ou **Kotlin** neste curso e com base nisso quero seguir o foco em uma.. O que vocês recomendam? e porquê?

Se tiver prós e contra de cada uma seria interessante também..

Se o vento bater certo estarei no final me candidatando para vagas de "dev apps Android" e acima de tudo estarei muito feliz em terminar/continuar o [app](https://github.com/doesangueorg/android) do meu projeto [doesangue](https://github.com/doesangueorg).

**Comment by jvrmaia** [comment]
Java ainda é o que domina no mercado mas o pessoal tem aderido bem ao Kotlin pois a linguagem ficou muito boa. Kotlin tbm já tá com um pé no mundo iOS [1]. Da uma olhada no curso da Loiane de Java [2]. Ela tem outros cursos além desse.

[1] https://kotlinlang.org/docs/reference/native-overview.html
[2] https://loiane.training/course/java-basico/

**Comment by JoseCage** [comment]
Valeu @jvrmaia.. Por acaso estava mesmo acompanhando o curso de Java da professora @loiane :heart:, ainda não terminei.

Vou fazer como a maioria então. Focar um pouco mais em Java por ter mais mercado mas manter um olho no Kotlin.

**Comment by lbighetti** [comment]
+1 **Kotlin**

Eu trabalhei com Java por 5 anos, eu acho Kotlin bem mais agradável, seguro e produtivo.
Meu stack preferido de Java (Spring Boot) agora tem suporte oficial pro Kotlin, e Android e AndroidStudio também.
Kotlin tem também features de programação funcional, muito mais do que Java, e eu gosto bastante disso.
Dá pra escrever builds scripts do Gradle em Kotlin agora também, e eu prefiro do que groovy (https://github.com/JetBrains/kotlin-examples/blob/master/gradle/hello-world/build.gradle)

Mas... talvez faça sentido pra você aprender e usar Java antes:

* pra entender quais os problemas que o Kotlin resolve, e porque é mais bacana que Java
* porque Kotlin é 100% interroperável com Java, então tudo que você aprender lá não é perdido, você carrega tudo e pode continuar usando.
* porque há uma tendência de fazer apps novos ou migrar apps Android para Kotlin, mas a maioria das vagas de Android ainda é em Java.

Eu pessoalmente acho Kotlin uma linguagem sensacional, e mesmo tendo parado de trabalhar com Java (to em Elixir/Erlang/Elm e um pouco de Python agora), eu continuo brincando e as vezes fazendo algum sideproject com Kotlin, as vezes vou em meetups também.

**Comment by JoseCage** [comment]
Boa @lbighetti.. Sinto isso também, que ainda será necessário conhecimento em Java ponta de ter mais mercado.

**Comment by woliveiras** [comment]
No mercado brasileiro, o @JoseCage conseguiria um emprego para trabalhar com mobile se aprender Kotlin, pessoal?

**Comment by jvrmaia** [comment]
No ES tem uma empresa que tem sempre procurado programadores mobile e usam Kotlin.

**Comment by lbighetti** [comment]
@woliveiras Sim, vejo vagas em São Paulo, Rio, Curitiba, Espírito Santo e Recife ativas no momento por exemplo.

Minha percepção é que as vagas de Kotlin estão crescendo exponencialmente, eu diria que hoje tem 10x mais vagas que o ano passado, não só tipo, 2x.

Com certeza ainda muito menos do que Java claro, mas a linguagem está em um crescimento brusco. Outro indicador disso é a IDE do Google passar a suportar Kotlin oficialmente e o maior framework de Java também passar a suportar Kotlin oficialmente.

Outro ponto é que, na minha concepção, se tornar proficiente em Kotlin é uma situação  interessantíssima de se estar porque existem **muito menos** experts em Kotlin do que em Java.

my 2 cents

**Comment by danilosilvadev** [comment]
Só pra comentar, existe a opção js com react native.

**Comment by JoseCage** [comment]
@danilosilvadev sou mais de backend(_embora js em vários casos entra no backend_) rsrs.. 

> Só pra comentar, existe a opção js com react native.

**Comment by JoseCage** [comment]
@woliveiras @lbighetti não sei no meu caso se vale "as estatísticas do Brasil" uma vez que sou Angolano, então a princípio seria para o nosso mercado mesmo, mas aí quem sabe aparece um remoto pra Brasil ou mesmo outro páis.

**Comment by lbighetti** [comment]
@josecage que legal ! Não sabia :)
Eu moro em Berlim na verdade, mas diria que minha percepção é geral, não só do Brasil ou aqui, eu acho que em todo lugar existe essa tendência, crescimento da linguagem e dos empregos, e que já é possível achar vagas em Kotlin porém muito menos do que o Java.

Não sei te dizer como é o mercado em Angola, ai você teria que dar uma pesquisada, mas o que eu posso te dizer, é que no caso das vagas serem primariamente de Java pra Android, você pode começar estudando isso mesmo que o tempo não vai ser perdido - todo seu conhecimento se Java vai ser útil no Kotlin, inclusive o Kotlin se torna bem fácil de aprender uma vez que você sabe Java (diferente de Scala e Clojure por exemplo).

Como você disse, remoto ou imigração são sim também possibilidades com Kotlin. Aqui em Berlim vejo vagas já por exemplo.

**Comment by woliveiras** [comment]
Só o pessoal internacional aqui! :joy:

**Comment by JoseCage** [comment]
Vou pesquisar melhor mesmo @lbighetti.. Valeu demais a contribuição. Mas se puder aplicar remoto vai valer também..

**Comment by linuxsoares** [comment]
Opa!
Eu acho que Kotlin vai crescer bastante, mas Java não vai morrer para o mundo mobile android. Sendo bem sincero eu acho que vale aprender Java, e quando estiver bem tranquilo com Java, trabalhando aprenda Kotlin.
Aprender as 2 é melhor do que 1!
Mas comece com Java!

**Comment by JoseCage** [comment]
Valeu @linuxsoares :smile:

**Comment by Morette** [comment]
Acho que Kotlin pega bastante da base de Java, então dá pra começar a aprender Java e depois passar para Kotlin, se quiser. A curva de aprendizado deve ser menor.
